File: backend-spring-boot/src/main/java/no/vicx/backend/user/validation/RecaptchaResponseVm.java
package no.vicx.backend.user.validation;

import com.fasterxml.jackson.annotation.JsonProperty;

// See https://developers.google.com/recaptcha/docs/verify

public record RecaptchaResponseVm(
        boolean success,
        @JsonProperty("challenge_ts")
        String challengeTimestamp,
        String hostname,
        @JsonProperty("error-codes")
        String[] errorCodes
) {
}

File: backend-ktor/src/main/kotlin/no/vicx/ktor/user/vm/RecaptchaResponseVm.kt
package no.vicx.ktor.user.vm

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RecaptchaResponseVm(
    val success: Boolean = false,
    @SerialName("challenge_ts") val challengeTimestamp: String = "",
    val hostname: String = "",
    @Suppress("ArrayInDataClass") @SerialName("error-codes") val errorCodes: Array<String> = emptyArray(),
)
